### PR TITLE
refactor(rxHideIfUkAccount): remove obsolete billing service

### PR DESCRIPTION
**Pull request by TheNando**
For https://jira.rax.io/browse/EOD-1435

I'll also be removing the directive from the Framework. It can be reviewed now, but release needs to be coordinated with the Billing release for the UK feature. @kyleondata 
 - [x] Dev
 - [x] Dev

### CODE REVIEW

File: src/components/rxApp/scripts/rxAppNav.js [hunk 1]
```diff
@@ -273,8 +273,7 @@ angular.module('encore.ui.rxApp')
  * environment is not necessary.
  *
  * Another technique for visibility is to use a predefined set of visibility
- * functions that exist in the framework. These include `rxPathParams` and
- * `rxHideIfUkAccount`.
+ * functions that exist in the framework. These include `rxPathParams`
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxAppNav.js]: I'd go with ```... in the framework (e.g. `rxPathParams`).```

**Comment by CITguy** [comment]
@TheNando, I'd call this merge capable, considering it's for the `2.x` branch.

**Comment by TheNando** [comment]
@CITguy that's true. I was erring on the side of caution as this must go out after the Billing UK feature release. I'm _pretty sure_ that will happen in 2 weeks, but just it might take longer. If you feel safe that 2.x won't happen in the new few weeks, I wouldn't have a problem with removing the No Merge label.

**Comment by CITguy** [comment]
@parlarjb, do you have any objections to merging this to `2.x`?

**Comment by CITguy** [comment]
LGTM

**Comment by parlarjb** [comment]
@CITguy No objections

**Comment by parlarjb** [comment]
LGTM

**Comment by CITguy** [comment]
Service removal documented in migration guide: https://github.com/rackerlabs/encore-ui/wiki/2.0-Migration-Guide#removed-services

**Pull request merged**